scripts/validate_metadata.py

```diff
@@ -0,0 +1,238 @@
+import sys
+from itertools import pairwise
+
+import yaml
+from datetime import datetime, timezone
+from pathlib import Path
+import logging
+
+from semver import Version
```

Comment:
`pyaml` and `node-semver` are not part of standard python library, so we need to include a `requirements.txt` for this script, otherwise those using this script will not know what versions of these packages they need to install to run the script. 

You should also install the dependencies in the github workflow prior to executing the script.